app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -38,7 +38,10 @@
 public class MainActivity extends AppCompatActivity
 {
     private static final String TAG = "LoyaltyCardLocker";
-    private Menu _menu;
+    private static final int MAIN_REQUEST_CODE = 1;
+
+    private Menu menu;
+    protected String filter = "";
```

Comment:
It took me a bit to figure out what this was for. This is to manipulate the search query so there can still be unit tests, correct? it is a little unconventional, though it does enable unit tests so I'll be more confident the feature will keep working. Let's go with it.

Reply:
Oh, you mentioned the filter variable in your comment. I looked at the code first... (: